atoma-state/src/state_manager.rs

```diff
@@ -619,6 +632,38 @@ impl AtomaState {
             .collect()
     }
 
+    /// Retrieves all stacks.
+    ///
+    /// This method fetches all stack records from the `stacks` table.
+    ///
+    /// # Returns
+    ///
+    /// - `Result<Vec<Stack>>`: A result containing either:
+    ///   - `Ok(Vec<Stack>)`: A vector of `Stack` objects representing all stacks.
+    ///   - `Err(AtomaStateManagerError)`: An error if the database query fails or if there's an issue parsing the results.
+    ///
+    /// - The database query fails to execute.
+    ///
+    /// # Example
+    ///
+    /// ```rust,ignore
+    /// use atoma_node::atoma_state::AtomaStateManager;
+    ///
+    /// async fn get_current_stacks(state_manager: &AtomaStateManager) -> Result<Vec<Stack>, AtomaStateManagerError> {
+    ///    state_manager.get_current_stacks().await
+    /// }
+    /// ```
+    #[instrument(level = "trace", skip_all)]
+    pub async fn get_all_stacks(&self) -> Result<Vec<Stack>> {
+        let stacks = sqlx::query("SELECT * FROM stacks")
```

Comment:
There can be a looooot of stacks, can we get them by timestamp and introduce a `limit`parameter ?

Reply:
Changed into new table, please take a look